src/lib/zAuction.ts

```diff
@@ -87,8 +103,41 @@ export async function getBidsForNft(
 	return bids;
 }
 
+export async function getBidsListForNfts(
+	baseApiUri: string,
+	contract: string,
+	tokenId: string[],
+) {
+	const nftIdsArray: string[] = [];
+
+	for (let i = 0; i < tokenId.length; i++) {
```

Comment:
You are also iterating `nftIdsArray` but with `tokenId` instead. Never iterate Array A with start/stop from array B.